src/main/java/core/basesyntax/Advanced.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Advanced {
+    public static void main(String[] args) {
+        FigureSupplier figureSupplier = new FigureSupplier();
+        Random rd = new Random();
+        Figure[] figures = new Figure[6];
+        for (int i = 0; i < figures.length; i++) {
+            if (i < figures.length / 2) {
+                figures[i] = figureSupplier.getRandomFigure(rd.nextInt(5) + 1);
```

Comment:
This method (getRandomFigure) should be as simple as possible for users. They should not know anything about amount of available figures. Random index of figure should be picked inside getRandomFigure method
```suggestion
                figures[i] = figureSupplier.getRandomFigure();
```